src/api/components/source.js

```diff
@@ -341,12 +341,18 @@ module.exports = class ComponentSource extends EntitySource {
     }
 
     isView(file) {
-        return anymatch([
+        const matchers = [
             `**/*${this.get('ext')}`,
             `!**/*${this.get('splitter')}*${this.get('ext')}`,
             `!**/*.${this.get('files.config')}.${this.get('ext')}`,
             `!**/${this.get('files.config')}.{js,json,yaml,yml}`
-        ], this._getPath(file));
+        ];
+
+        if (this.get('exclude')) {
```

Comment:
I think we need a type check here. Someone could add anything in the config here.

Can we also open an issue to update the docs before this gets merged?